CMakeLists.txt

```diff
@@ -1,16 +1,21 @@
-cmake_minimum_required(VERSION 3.16...3.25)
+cmake_minimum_required(VERSION 3.16...3.27)
 
 include(cmake/RecordCmdLine.cmake)
 
-project(VecCore LANGUAGES C CXX VERSION 0.8.1
+option(CUDA    "Enable support for CUDA")
```

Comment:
I'm not a big fan of moving `option`s before `project`. Can we keep the `cmake/CUDA.cmake` file and just call `enable_language(CUDA)` there?